### ISSUE TITLE
Create/log workout - limit workout name field

### ISSUE THREAD

**Issue by joshwouden** [ifit/Enhancements#86]
#### Enhancement Details

---

**Description**
We currently allow very long names when a user is creating or logging a workout. However, names longer than ~30 characters cannot be seen anywhere on the site and cause some minor layout issues. 

We may want to consider limiting the names to 29 characters, since names longer than this will be truncated/cut off, create layout issues, and can't even be seen anywhere anyways.

**Screen-caps**
I made a workout that was 99 characters long
![image](https://cloud.githubusercontent.com/assets/7559375/5846151/c9e7fe44-a17e-11e4-8082-98dc937fbdf5.png)

On profile and community the long names go completely off the tile:
![image](https://cloud.githubusercontent.com/assets/7559375/5882776/ba67a2ee-a309-11e4-92ee-cbe00d0d201e.png)

The workout details page currently shows 24 characters, but we have space for a bit more:
![image](https://cloud.githubusercontent.com/assets/7559375/5882139/00edcf1e-a304-11e4-9f66-94715b3a6efa.png)

On the Workouts list, we can fit 29:
![image](https://cloud.githubusercontent.com/assets/7559375/5882116/cda3014c-a303-11e4-8d85-e340ea47c54e.png)

28 on the Android console:
![image](https://cloud.githubusercontent.com/assets/7559375/5882701/f00b17c4-a308-11e4-98a2-f2c0cca0560b.png)

26 on the dashboard:
![image](https://cloud.githubusercontent.com/assets/7559375/5882211/a24da9ec-a304-11e4-80d6-9bd83dbd1e85.png)

21 on the Schedule:
![image](https://cloud.githubusercontent.com/assets/7559375/5882177/4585a61a-a304-11e4-8ca1-fb8cc1bf93b9.png)

11 on the Calendar:
![image](https://cloud.githubusercontent.com/assets/7559375/5882186/67305422-a304-11e4-8455-864b25f8a5fc.png)

Technically we could show up to 63 on the workout creation screen, but it doesn't look very nice:
![image](https://cloud.githubusercontent.com/assets/7559375/5882086/9223cb42-a303-11e4-8544-53981937bea2.png)

**tl:dr**
Limit this field to ~29 characters:
![image](https://cloud.githubusercontent.com/assets/7559375/5882793/e932dc7e-a309-11e4-9e12-4a4074e873ed.png)

**Reproduction Steps**
Create a workout with a really long name
Note how the name appears on different parts of the site

**Notes**
@joeyhdez10
